### ISSUE TITLE
textDocument/formatting does not pass model options

### ISSUE THREAD

**Issue by rubenfiszel** [TypeFox/monaco-languageclient#356]
I might be doing something wrong but the languageClient does correctly send a `textDocument/formatting` payload to the lsp server whenever the action: `editor.getAction('editor.action.formatDocument').run()`, but the options it pass are an empty object: `{}` when I would have expected it to be the model options which have been set to `model.updateOptions({ tabSize: 4, insertSpaces: true })`. This result in deno lsp returning an error because it expects tabSize to be there.

**Comment by CGNonofr** [comment]
I think you need to implement the `configurations` field in the workspace service

**Comment by rubenfiszel** [comment]
@CGNonofr not sure to fully follow.

I tried inside the clientOptions: 
```
						middleware: {
							workspace: {
								configuration: (params, token, configuration) => {
									return [
										{
											tabSize: 4,
											enable: true,
											insertSpaces: true
										}
									]
								}
							}
						}
```

but that did not seem to do the trick

**Comment by CGNonofr** [comment]
I think it's 'editor.tabSize'. Btw you should probably implement it in the service itself (Services.install)

**Comment by rubenfiszel** [comment]
Fixed by #359
